src/python/pants/auth/basic_auth.py

```diff
@@ -84,7 +84,7 @@ def authenticate(self, provider: str, creds: Optional[BasicAuthCreds] = None, co
       auth = requests.auth.HTTPBasicAuth(creds.username, creds.password)
     else:
       auth = None  # requests will use the netrc creds.
-    response = requests.get(url, auth=auth)
+    response = requests.get(url, auth=auth, headers={'User-Agent': f"pants/v{VERSION}"})
```

Comment:
Every other string in this file is ', consider keeping the f-string added here in line with that precedent.